### PR TITLE
Add possibility to set custom scale factor for iOS Simulator in xcuitest driver

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This PR is based on https://github.com/appium/appium-ios-simulator/pull/74

### CODE REVIEW

File: lib/desired-caps.js [hunk 1]
```diff
@@ -33,6 +33,9 @@ let desiredCapConstraints = _.defaults({
   tapWithShortPressDuration: {
     isNumber: true
   },
+  scaleFactor: {
+    isString: true
```
Comment (imurchie) [inline lib/desired-caps.js]: Should this be a number? What are the valid values of the `scaleFactor`? If it is closely circumscribed the validation might need to be done manually, in [validateDesiredCaps](https://github.com/appium/appium-xcuitest-driver/blob/master/lib/driver.js#L534)

File: lib/driver.js [hunk 2]
```diff
@@ -162,6 +162,9 @@ class XCUITestDriver extends BaseDriver {
     }
 
     let { device, udid, realDevice } = await this.determineDevice();
+    if (!realDevice && this.opts.scaleFactor !== undefined) {
```
Comment (imurchie) [inline lib/driver.js]: Can this new bit be moved below the logging and setting of the device information? Separating the getting of device and using what was retrieved makes telling what is going on a bit more difficult.

File: lib/desired-caps.js (hunk 1 above)
Comment (mykola-mokhnach) [inline lib/desired-caps.js]: I've implemented value validation in https://github.com/appium/appium-ios-simulator/pull/74  :

```javascript
 setScaleFactor (newScaleFactor) {
    let supportedScales = ['1.0', '0.75', '0.5', '0.33', '0.25'];
    if (supportedScales.indexOf(newScaleFactor) < 0) {
      let msg = `Only "${supportedScales}" values are supported as scale factors. "${newScaleFactor}" is passed instead.`;
      log.errorAndThrow(msg);
    }
    this.scaleFactor = newScaleFactor;
  }
```

Comment (imurchie) [inline lib/desired-caps.js]: Ok. It is early enough in the flow to kill the session creation when this fails. Thanks.

File: lib/driver.js (hunk 2 above)
Comment (mykola-mokhnach) [inline lib/driver.js]: Is it better now?

**Comment by imurchie** [comment]
Yes! Thanks.

**Comment by mykola-mokhnach** [comment]
@imurchie shall I also add the documentation for this parameter to the main Appium lib?

**Comment by imurchie** [comment]
Yes, unfortunately. In the section for XCUITest-only (https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/caps.md#ios-only-using-xcuitest)

**Pull request merged**